agent/consul/gateways/controller_gateways.go

```diff
@@ -0,0 +1,39 @@
+package gateways
+
+import (
+	"context"
+	"github.com/hashicorp/consul/agent/consul/controller"
+	"github.com/hashicorp/consul/agent/consul/fsm"
+	"github.com/hashicorp/consul/agent/consul/state"
+	"github.com/hashicorp/consul/agent/consul/stream"
+	"github.com/hashicorp/go-hclog"
+)
+
+//baseReconciler contains fields shared across multiple controller.Reconciler implementations in the gateways package
+type baseReconciler struct {
```

Comment:
Main thing here is that I'm not really sure we need this -- I'd imagine that the various reconcilers are going to leverage different fields at some point and that aside from maybe the route reconcilers, the logic is going to be fairly different, so there's not going to be much else on this struct...

So, personally, I'd prefer if we just dropped the struct from this and declared the fields directly on the other reconciler structs rather than embedding this. Aside from that, this looks fine.

Reply:
done